clutel/clutel.go

```diff
@@ -0,0 +1,33 @@
+package clutel
+
+import (
+	"context"
+
+	"github.com/alcionai/clues/internal/node"
+	"github.com/alcionai/clues/internal/stringify"
+)
+
+// AddToOTELHTTPLabeler adds key-value pairs to both the current
+// context and the OpenTelemetry HTTP labeler, but not the current
+// span.  The labeler will hold onto these values until the next
+// request arrives at the otelhttp transport, at which point they
+// are added to the span for that transport.
+//
+// The best use case for this func is to wait until the last wrapper
+// used to handle a http.Request.Do() call.  Add your http request
+// details (url, payload metadata, etc) at that point so that they
+// appear both in the next span, and in any errors you handle from
+// that wrapper.
+func AddToOTELHTTPLabeler(
+	ctx context.Context,
+	name string,
+	kvs ...any,
+) context.Context {
+	nc := node.FromCtx(ctx)
+
+	return node.EmbedInCtx(ctx, nc.AddValues(
+		stringify.Normalize(kvs...),
+		node.DoNotAddToSpan(),
+		node.AddToOTELHTTPLabeler(ctx),
```

Comment:
should using a labeler automatically disable adding to the current span? I'm sort of also trying to determine if there could be ordering issues in the future (yea, my usual nit about varargs for options)